core/src/main/java/com/microsoft/dagx/system/DefaultServiceExtensionContext.java

```diff
@@ -48,13 +43,13 @@ public DefaultServiceExtensionContext(TypeManager typeManager, Monitor monitor,
         this.monitor = monitor;
         this.serviceLocator = serviceLocator;
         // register as services
-        this.services.put(TypeManager.class, typeManager);
-        this.services.put(Monitor.class, monitor);
+        services.put(TypeManager.class, typeManager);
+        services.put(Monitor.class, monitor);
     }
 
     public void initialize() {
         configurationExtensions = loadExtensions(ConfigurationExtension.class, false);
-        configurationExtensions.forEach(ext -> ext.initialize(monitor));
+        configurationExtensions.forEach(ext -> ext.initialize(this));
```

Comment:
This looks a little bit strange. The DefaultServiceExtensionContext has a list of object, which hold a reference to the DefaultServiceExtensionContext

Reply:
actually it's the service context that initializes configuration extensions, but some of them actually need the context, e.g. to obtain settings. see `AzureVaultExtension`

Reply:
but VaultExtension gets loaded later on (or do you want to make it a ConfigurationExtension?). 
This is init of the ServiceExtensionContext and as init has not been completed it passes itself as a potentially just partly initialized object to an extension.

Reply:
you have are right about the partly initialised object. that needs to be changed.